### PR TITLE
Tushar leetcode

### PR DESCRIPTION
**Pull request by tushar8049**

### CODE REVIEW

File: 0043. Multiply String/Solution.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+//
+//Output: "56088"
+//
+//Note:
+//
+//The length of both num1 and num2 is < 110.
+//Both num1 and num2 contain only digits 0-9.
+//Both num1 and num2 do not contain any leading zero, except the number 0 itself.
+//You must not use any built-in BigInteger library or convert the inputs to integerdirectly.
+
```
Comment (pradhan1234) [inline 0043. Multiply String/Solution.java:20]: Looks Great! Do add space time complexity of the solution.

File: 0043. Multiply String/Solution.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            }
+        }
+
+        StringBuilder result = new StringBuilder();
+
+        for(int i = 0; i < d.length; i++){
+            int carry = d[i] / 10;	// Carry Forward string
+            int rem = d[i] % 10;	//Reminder that will be as it is in the output string
+            if(i+1 < d.length){
+                d[i+1] += carry;    //Adding Cary forward String
```
Comment (pradhan1234) [inline 0043. Multiply String/Solution.java:52]: typo here

File: 0043. Multiply String/Solution.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            }
+            result.insert(0, rem); // reminder inserted at the begining everytime because we are accessing it from the end and we don't wanna reverse it again after building the string.
```
Comment (pradhan1234) [inline 0043. Multiply String/Solution.java]: remainder*

File: 0043. Multiply String/Solution.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    public static String multiply(String num1, String num2) {
+        // Reverse becasue it becomes easy for accessing the elements from the end of the number string
+        String n1 = new StringBuilder(num1).reverse().toString();
+        String n2 = new StringBuilder(num2).reverse().toString();
+
+        // store the basic multiplication of numbers maybe in 2 digits which can be later carry forwarded to a different string.
+        int[] d = new int[ n1.length() + n2.length() ];
+
+        for(int i = 0; i < n1.length(); i++){
```
Comment (pradhan1234) [inline 0043. Multiply String/Solution.java:40]: brief comment about what's going on here

**Review by pradhan1234 — CHANGES_REQUESTED** [review]

**Review by pradhan1234 — APPROVED** [review]

**Review by tushar8049 — COMMENTED** [review]
@pradhan1234 Can you look into this!

File: 0232. Implementing Queue using Stack/MyQueue.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
+class MyQueue {
+    /** Two stacks to maintain the queue functionoing */
+    private Stack<Integer> s1 = new Stack<>();
+    private Stack<Integer> s2 = new Stack<>();
+
+    /** Initialize your data structure here. */
+    public MyQueue() {
+    }
+
```
Comment (pradhan1234) [inline 0232. Implementing Queue using Stack/MyQueue.java:11]: Looks Great, just nit picks.
Better to declare stack s1 and s2 above constructor and initialize them using the constructor.
you can consider moving new stack<>() inside constructor.

File: 0232. Implementing Queue using Stack/MyQueue.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        return s2.peek();
+    }
+
+    /** Returns whether the queue is empty. */
+    /** We need to check both the queues for emptyness. */
+    public boolean empty() {
+        return s1.isEmpty() && s2.isEmpty();
+    }
+}
+
```
Comment (pradhan1234) [inline 0232. Implementing Queue using Stack/MyQueue.java:50]: another small point
for comments over push, pop, peek and empty use one multiline comment to contain the comments

**Review by pradhan1234 — CHANGES_REQUESTED** [review]

File: 0080. Remove Duplicates from Sorted Array II/Solution.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
+/** Given a sorted array nums, remove the duplicates in-place such that duplicates appeared at most twice and return the new length.
+
+ Do not allocate extra space for another array, you must do this by modifying the input array in-place with O(1) extra memory. */
+
```
Comment (pradhan1234) [inline 0080. Remove Duplicates from Sorted Array II/Solution.java:4]: Space Time Complexity

**Review by pradhan1234 — CHANGES_REQUESTED** [review]

**Review by pradhan1234 — APPROVED** [review]

**Pull request merged**